src/components/calcite-time-picker/calcite-time-picker.tsx

```diff
@@ -0,0 +1,837 @@
+        break;
+      case this.minuteEl:
+        switch (event.key) {
+          case "ArrowLeft":
+            this.setFocus("hour");
+            break;
+          case "ArrowRight":
+            if (this.step !== 60) {
+              this.setFocus("second");
+            } else if (this.hourDisplayFormat === "12") {
+              this.setFocus("meridiem");
+            }
+            break;
+        }
+        break;
+      case this.secondEl:
+        switch (event.key) {
+          case "ArrowLeft":
+            this.setFocus("minute");
+            break;
+          case "ArrowRight":
+            if (this.hourDisplayFormat === "12") {
+              this.setFocus("meridiem");
+            }
+            break;
+        }
+        break;
+      case this.meridiemEl:
+        switch (event.key) {
+          case "ArrowLeft":
+            if (this.step !== 60) {
+              this.setFocus("second");
+            } else {
+              this.setFocus("minute");
+            }
+            break;
+        }
+        break;
+    }
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  //--------------------------------------------------------------------------
+
+  @Method()
+  async setFocus(target: TimeFocusId): Promise<void> {
+    this[`${target || "hour"}El`]?.focus();
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  // --------------------------------------------------------------------------
+
+  private decrementHour = (): void => {
+    const newHour = !this.hour ? 0 : this.hour === "00" ? 23 : parseInt(this.hour) - 1;
+    this.setTime("hour", newHour);
+  };
+
+  private decrementMeridiem = (): void => {
+    const newMeridiem = this.meridiem === "PM" ? "AM" : "PM";
+    this.setTime("meridiem", newMeridiem);
+  };
+
+  private decrementMinuteOrSecond = (key: MinuteOrSecond): void => {
+    let newValue;
+    if (isValidNumber(this[key])) {
+      const valueAsNumber = parseInt(this[key]);
+      if (valueAsNumber === 0) {
+        newValue = 59;
+      } else {
+        newValue = valueAsNumber - 1;
+      }
+    } else {
+      newValue = 59;
+    }
+    this.setTime(key, newValue);
+  };
+
+  private decrementMinute = (): void => {
+    this.decrementMinuteOrSecond("minute");
+  };
+
+  private decrementSecond = (): void => {
+    this.decrementMinuteOrSecond("second");
+  };
+
+  private buttonActivated(event: KeyboardEvent): boolean {
+    const isKeyboardEvent = event instanceof KeyboardEvent;
+    const enterPressed = isKeyboardEvent && event.key === "Enter";
+    const spacebarPressed = isKeyboardEvent && (event.key === " " || event.key === "Spacebar");
+    if (spacebarPressed) {
+      event.preventDefault();
+    }
+    return enterPressed || spacebarPressed;
+  }
+
+  private focusHandler = (event: FocusEvent): void => {
+    this.activeEl = event.target as HTMLSpanElement;
```

Comment:
I think in this case it'll be the same, but it's better to use `currentTarget` (element processing the event or with the listener attached) than `target` to guarantee the element.